### ISSUE TITLE
Support "fail fast" & rollback

### ISSUE THREAD

**Issue by stephen-soltesz** [m-lab/scraper#71]
There was a long delay before the issue with oauth2client manifested on mlab-staging.

Support "fail fast". It should be possible to run some preliminary operation on start up that exercises all the authentication necessary (not necessarily the same credentials, but the same auth mechanism) used later in the process.

The sooner we fail the easier it will be to take advantage of kubernetes rollout and rollback mechanisms.

**Comment by pboothe** [comment]
I agree. However, those code paths were working fine for the previous 12 hours. So the bug that manifested was not "unexecuted code turns out to not work when its time comes up", but instead "code that has been executing successfully stops working".
